src/components/Gallery.tsx

```diff
@@ -0,0 +1,101 @@
+import ImgixAPI, { APIError } from 'imgix-management-js';
+import { Component } from 'react';
+import Imgix from 'react-imgix';
+import { SourceProps } from './Dialog';
+
+interface GalleryProps {
+  selectedSource: Partial<SourceProps>;
+  imgix: ImgixAPI;
+}
+
+interface GalleryState {
+  imgix: ImgixAPI;
+  fullUrls: Array<string>;
+}
+
+export default class Gallery extends Component<GalleryProps, GalleryState> {
+  constructor(props: GalleryProps) {
+    super(props);
+
+    this.state = {
+      imgix: props.imgix,
```

Comment:
This is an anti-pattern, it's better to use `this.props.imgix` instead of assigning it to state

Reply:
ah good catch